source/tutorials/broadcom.rst

```diff
@@ -0,0 +1,203 @@
+#. See if your device is listed on the support matrix of this driver:
+   https://wireless.wiki.kernel.org/en/users/Drivers/b43#list_of_hardware
+
+#. The firmware for cards supported by the *b43* and *b43legacy* drivers
+   usually needs to be sourced and installed manually. Continue reading the
+   `Firmware`_ section of this document.
+
+
+broadcom-wl
+===========
+
+*broadcom-wl*, also known as *broadcom-sta* or *wl*, is the proprietary closed
+source driver from Broadcom and tends to work only for older devices. As such,
+it is not part of the Linux kernel and has to be built as an out-of-tree
+kernel module.
+
+#. See if your device is supported and download the Linux 64-bit driver from
+   Broadcom's website or another source:
+   https://www.broadcom.com/support/download-search?pg=&pf=Wireless+LAN+Infrastructure
+   
+#. Extract the downloaded archive into a separate folder. For example:
+   
+   .. code-block:: bash
+
+      mkdir ./broadcom-wl/
+      tar xvf ./hybrid-v35_64-nodebug-pcoem-6_30_223_271.tar.gz -C broadcom-wl/
+      cd ./broadcom-wl/
+   
+#. :ref:`Install the DKMS bundle <kernel-modules-dkms-install-begin>` for your
+   kernel. DKMS provides the framework to automatically rebuild the wl driver
+   against new kernels versions from |CL| updates.
+
+#.  In the extracted driver directory, create a :file:`dkms.conf` file based
+    the contents below to provide DKMS information about how to build and
+    install the kernel module. This example uses version *6.30.223.271*.
+
+    .. code-block:: bash
+
+       cat <<'EOF' >> dkms.conf
+       PACKAGE_NAME=broadcom-wl
+       PACKAGE_VERSION=6.30.223.271
+       MAKE="make KBASE=/lib/modules/${kernelver}"
+       CLEAN="make KBASE=/lib/modules/${kernelver} clean"
+       BUILT_MODULE_NAME=wl
+       DEST_MODULE_LOCATION=/kernel/drivers/net/wireless
+       AUTOINSTALL=yes
+       EOF
+
+#. Run the :command:`dkms` commands to add the broadcom-wl module to the dkms
+   tree, build it, and install it. This example uses version *6.30.223.271*.
+
+   .. code-block:: bash
+
+      dkms add -m broadcom-wl -v 6.30.223.271
+      dkms build -m broadcom-wl -v 6.30.223.271
+      dkms install -m broadcom-wl -v 6.30.223.271
+
+#. Blacklist all other variations of Broadcom drivers from loading otherwise
+   they can conflict and cause problems.
+
+   .. code-block:: bash
+
+      sudo mkdir -p /etc/modprobe.d/
+
+      sudo tee /etc/modprobe.d/broadcom.conf > /dev/null <<'EOF'
+      blacklist b43
+      blacklist b43legacy
+      blacklist ssb
+      blacklist bcm43xx
+      blacklist brcm80211
+      blacklist brcmfmac
+      blacklist brcmsmac
+      blacklist bcma
+      EOF
+
+#. Reboot the system and check that the module is loaded and working. If not,
+   try manually updating dependencies and loading the module.
+
+   .. code-block:: bash
+
+      depmod -a
+      modprobe wl
+
+
+Firmware
+********
+
+In addition to device drivers, firmware is required that gets loaded onto the
```

Comment:
In addition to drivers, devices require firmware to be loaded onto the device directly.